src/ParseServer.js

```diff
@@ -59,6 +64,82 @@ class ParseServer {
   constructor(options: ParseServerOptions) {
     // Scan for deprecated Parse Server options
     Deprecator.scanParseServerOptions(options);
+
+    // START
```

Comment:
Did you get a chance to check this?

I was looking into it and as I mentioned in one of my earlier messages, this part of the code wasn't the problem, I wrote this code as well (although I was going about it in a different way), the problem is the fact that the values are of different types. For example,
- There are 2 keys called fileUpload & push which according to definition are of type any. When it receives an object this code will fail which it shouldn't because the value can be anything. (We'll have to somehow keep track if the value 
- Then there is a variable called rateLimit, which according to definition expects an array but in the function it is defined in, it excepts both, a value of type RateLimitOptions and a list of RateLimitOptions. This code doesn't handle these cases. (I think arrays in general are not handled here) 
	- (Although it can be added with one more else condition but then another case arises where the user might be giving a wrong type and will get an error for some key which might be misleading, which makes me think that without type checking doing this might not be such a good idea.)
- Another thing is adapter that are failing. In the original solution as well I wasn't check the keys in adapters, but now thinking about it, I think if we go with original approach we might be able to call the validation Function from adapter Constructors. It will be difficult to determine the correct adapter to validate the keys with current approach.

Reply:
Responding [here](https://github.com/parse-community/parse-server/pull/8987#issuecomment-2024161596)